### DISCUSSION TITLE
How to get the full URL inside a server endpoint.

### DISCUSSION THREAD

**Discussion by LucasCoderT** [nuxt/nuxt#16714] [Questions] [answered]
Hi,

I'm wondering how would I be able to get the full URL from within a server endpoint. I can only seem to get the endpoint itself, not the full URL.

Thanks

**Comment by sevillaarvin** [comment]
You have to pass the domain through env variables or hard coded in nuxt.config, then retrieve at runtime.

**Reply by LucasCoderT** [reply]
I'm more asking because I'm using supabase auth, and it's passing the oauth callback using `#` to the url instead of a query param. 
eg:
`https://localhost:3000/api/callback#authkey=...`

So I'm trying to find a way to parse that on a nuxt server endpoint.  I figured if I could just get the full url being visited I could just parse it manually as I found no other indication/possibility  in the h3 docs about parsing params by `#`

**Reply by sevillaarvin** [reply]
How about getting the `HOST` header from the request headers? `event.req.headers.host`

**Comment by TheAlexLichter** [comment, marked as answer]
The fragment part of the URL (`#fragment=ajiaosdjs`) is only available in the browser and will not be sent to the server.
Instead, use query params (and then e.g. `getQuery`) to provide the key.
